src/imports/Components/1.3/OrientationHelper.qml

```diff
@@ -118,7 +118,7 @@ Item {
 
             var availableHeight = orientationHelper.parent.height;
             if (d.stateAngle === 0 && UbuntuApplication.inputMethod.visible && anchorToKeyboard)
-                availableHeight -= UbuntuApplication.inputMethod.keyboardRectangle.height;
+                availableHeight -= UbuntuApplication.inputMethod.keyboardRectangle.height / Screen.devicePixelRatio;
```

Comment:
Interesting that we take the keyboardRectangle given to us and need to modify it... might a better solution be to fix keyboardRectangle in UbuntuApplication? Or is it only wrong in select cases?

Reply:
Maybe that would work, too.
I think this problem appeared after the scaling was introduced (and only for apps that use scaling like morph, for others the factor is 1 I think.)
I've put the change as done in this example for morph before:
https://github.com/ubports/morph-browser/commit/ce68185b304db5f9de51fae4678d421a0037f077#diff-1c9bca71438f627d69c9571b4f62bd76
"units.gridUnit / 8" is the same as Screen.devicePixelRatio here.
So if we change UbuntuApplication.inputMethod.keyboardRectangle.height we have to make sure to adapt / revert all changes that use that parameter.
Not sure if other apps might need the absolute value of the keyboard rectangle...

Reply:
ah, so I think `UbuntuApplication.inputMethod.keyboardRectangle` is just an alias to `Qt.inputMethod.keyboardRectangle` which we cannot change since it comes directly as static QML value, can we ?

Reply:
https://github.com/ubports/ubuntu-ui-toolkit/blob/eb36fcaad14e28b9e73fe4f4d5c73fbe81922855/src/UbuntuToolkit/ucapplication.cpp here the property initialized with `QGuiApplication::inputMethod()` so I think keyboardRectangle is only readonly here.

Reply:
Okay, no problem then. If this creates the expected behavior (which it seems to), let's go for it!

Reply:
ok : ) Should be fine, at least on my phone (patched locally with that since Nov) I've seen no side effects in other apps.